hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/OmKeyInfo.java

```diff
@@ -658,18 +655,25 @@ public Builder setFile(boolean isAFile) {
     }
 
     public Builder setTags(Map<String, String> tags) {
-      this.tags.clear();
-      addAllTags(tags);
+      if (tags == null || tags.isEmpty()) {
+        this.tags = MapBuilder.empty();
+      } else if (tags instanceof ImmutableMap) {
+        this.tags = MapBuilder.of((ImmutableMap<String, String>) tags);
+      } else {
+        this.tags = MapBuilder.copyOf(tags);
+      }
```

Comment:
Please do not create new `MapBuilder`.

```suggestion
      this.tags.set(tags);
```